drivers/watchdog/iwdg_stm32.c

```diff
@@ -101,60 +108,39 @@ static int iwdg_stm32_set_config(struct device *dev,
 	/* Neither STM32F1X nor STM32F4 series supports window option. */
 	LL_IWDG_SetWindow(iwdg, 0x0FFF);
 #endif
-
-	tickstart = k_uptime_get_32();
-
-	while (LL_IWDG_IsReady(iwdg) == 0) {
```

Comment:
Why do you remove this part?

Reply:
Exactly, this part is present in the HAL stm32f3xx_hal_iwdg.h file in method **HAL_IWDG_Init**. Purpose is clear: Wait for register to be updated. Should stay in my opinion.

Reply:
the old version enable IWDG(LL_IWDG_Enable) first and then do iwdg_stm32_set_config. So it's ok to wait IDDG ready.

But the new API doesn't enable IWDG when install timeout, it is enabled when setup. 

I don't think it's ok waiting IWDG ready even if it is doesn't enabled.